src/core/execution_context/in_place_exec_ctxt.cc

```diff
@@ -137,6 +137,13 @@ void InPlaceExecutionContext::TearDownIterationAll(
 
   auto* rm = Simulation::GetActive()->GetResourceManager();
   rm->EndOfIteration();
+  // Load balancing destroys the synchronization between the simulation and the
+  // environment. We mark the environment aus OutOfSync such that we can update
+  // the environment before acessing it again.
+  // ToDiscuss(Lukas): better place that function to rm->AddAgent,
+  // rm->RemoveAgent?
+  auto* env = Simulation::GetActive()->GetEnvironment();
```

Comment:
I think the comment is not clear.
The environment has to be updated at least once per iteration regardless of load balancing.
Maybe redundant with the call to `ForcedUpdate` in the tear down operation.